src/plugins/MediaManager.js

```diff
@@ -363,4 +365,12 @@ module.exports = class MediaManager {
             this.stopVideoStreaming();
         }
     }
+
+    /**
+     * Plugin destructor, responsible for removing all callbacks & bindings so that things are garbage-collected
+     */
+    destroy() {
+        this.microphonePermissionObject?.removeEventListener('change', this.onMicrophonePermissionChange.bind(this));
```

Comment:
Maybe we can keep an array with all permissions object (or all fn to call for cleanup) and iterate over ? This way if we add new listener to cleanup all we need to do it's to add the clean up fn to the array

Reply:
Maybe the array can be shared accross plugin, so throught DeviceRender we can clean up all things

Reply:
As discussed on slack, I think this is out of scope for this PR. I'll try a POC PR based on this one, just to see if this is too big of a change or if it's OK.

Reply:
draft PR here: https://github.com/Genymobile/genymotion-device-web-player/pull/51